tests/reference/cpp-expr2-09c05ad.stdout

```diff
@@ -25,8 +25,8 @@ void test_boolOp()
     a = a && b;
     b = a || true;
     a = a || b;
-    a = a && b == b;
-    a = a && b != b;
+    a = a && (int)(b) == (int)(b);
+    a = a && (int)(b) != (int)(b);
```

Comment:
`b` should not be explicitly converted to `int`, right??

Reply:
yeah, I see, there are many such similar issues with `cpp` `codegen`. We can try to address all of them together.

Reply:
I think in this particular place it's a bug in ASR, there should not be a cast to int.

Reply:
Fixed that.